core/Decode.cpp

```diff
@@ -148,20 +172,125 @@ namespace olympia
             // Send instructions on their way to rename
             for (uint32_t i = 0; i < num_decode; ++i)
             {
-                const auto & inst = fetch_queue_.read(0);
-                insts->emplace_back(inst);
-                inst->setStatus(Inst::Status::DECODED);
-
-                if (fusion_enable_)
-                {
-                    uids.push_back(inst->getMavisUid());
+                if(uop_queue_.size() > 0){
+                    const auto & inst = uop_queue_.read(0);
+                    insts->emplace_back(inst);
+                    inst->setStatus(Inst::Status::DECODED);
+                    ILOG("From UOp Queue Decoded: " << inst);
+                    uop_queue_.pop();
+                }
+                else{
+                    auto & inst = fetch_queue_.read(0);
+                    
+                    // for vector instructions, we block on vset and do not allow any other processing of instructions until the vset is resolved
+                    // optimizations could be to allow scalar operations to move forward until a subsequent vector instruction is detected
+                    // or do vset prediction
+
+                    // mavis_id 214 -> vsetvl
+                    // mavis_id 216 -> vsetivli
+                    // TODO: change back from getMnemonic() to getInstructionUniqueID() where the ID doesn't change
+                    if(inst->getOpCodeInfo()->getInstructionUniqueID() == mavis_vsetivli_uid_){
+                        // vsetivli with immediates, we can set at decode and continue to process instruction group, no vset stall
+                        VCSRs_.lmul = inst->getLMUL();
+                        VCSRs_.vl = inst->getVL();
+                        VCSRs_.vta = inst->getVTA();
+                        VCSRs_.sew = inst->getSEW();
+                    }
+                    else if(inst->isVset() && (inst->getSourceOpInfoList()[0].field_value != 0 || inst->getOpCodeInfo()->getInstructionUniqueID() == mavis_vsetvl_uid_))
```

Comment:
please break statement into multiple lines for legibility

Reply:
Clang format applied in [b8028a3](https://github.com/riscv-software-src/riscv-perf-model/pull/167/commits/b8028a38dd6f24d24687a0984feca2d5f382d1a9)